plugins/inputs/couchbase/couchbase.go

```diff
@@ -0,0 +1,89 @@
+package couchbase
+
+import (
+	couchbase "github.com/couchbase/go-couchbase"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"sync"
+)
+
+type Couchbase struct {
+	Servers []string
+}
+
+var sampleConfig = `
+  ## specify servers via a url matching:
+  ##  [protocol://][:password]@address[:port]
+  ##  e.g.
+  ##    http://couchbase-0.example.com/
+  ##    http://admin:secret@couchbase-0.example.com:8091/
+  ##
+  ## If no servers are specified, then localhost is used as the host.
+  ## If no protocol is specifed, HTTP is used.
+  ## If no port is specified, 8091 is used.
+  servers = ["http://localhost:8091"]
+`
+
+func (r *Couchbase) SampleConfig() string {
+	return sampleConfig
+}
+
+func (r *Couchbase) Description() string {
+	return "Read metrics from one or many couchbase clusters"
+}
+
+// Reads stats from all configured clusters. Accumulates stats.
+// Returns one of the errors encountered while gathering stats (if any).
+func (r *Couchbase) Gather(acc telegraf.Accumulator) error {
+	if len(r.Servers) == 0 {
+		r.gatherServer("http://localhost:8091/", acc)
+		return nil
+	}
+
+	var wg sync.WaitGroup
+
+	var outerr error
+
+	for _, serv := range r.Servers {
+		wg.Add(1)
+		go func(serv string) {
+			defer wg.Done()
+			outerr = r.gatherServer(serv, acc)
+		}(serv)
+	}
+
+	wg.Wait()
+
+	return outerr
+}
+
+func (r *Couchbase) gatherServer(addr string, acc telegraf.Accumulator) error {
+	client, err := couchbase.Connect(addr)
+	if err != nil {
+		return err
+	}
+	pool, err := client.GetPool("default")
```

Comment:
Here you can make the client and pool getting a "pluggable" function. That way you can write unit tests that plugin a function that returns a mocked pool.

Shouldn't be too hard to create the mocked return data because the go data structures are based off JSON, so you should be able to use json.Marshal with some sample json.